File: collect_app/src/test/java/org/odk/collect/android/support/WidgetTestActivity.java
package org.odk.collect.android.support;

import androidx.fragment.app.FragmentActivity;
import androidx.lifecycle.LifecycleOwner;

import org.odk.collect.android.audio.ScreenContext;

public class WidgetTestActivity extends FragmentActivity implements ScreenContext {

    @Override
    public FragmentActivity getActivity() {
        return this;
    }

    @Override
    public LifecycleOwner getViewLifecycle() {
        return this;
    }
}

File: collect_app/src/test/java/org/odk/collect/android/support/WidgetTestActivity.kt
package org.odk.collect.android.support

import android.view.ContextMenu
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import org.odk.collect.android.application.CollectComposeThemeProvider

class WidgetTestActivity : AppCompatActivity(), CollectComposeThemeProvider {
    @JvmField
    val viewsRegisterForContextMenu = mutableListOf<View>()

    @JvmField
    val viewsWithShownContextMenu = mutableListOf<View>()

    override fun registerForContextMenu(view: View) {
        super.registerForContextMenu(view)
        viewsRegisterForContextMenu.add(view)
    }

    override fun onCreateContextMenu(menu: ContextMenu, v: View, menuInfo: ContextMenu.ContextMenuInfo?) {
        super.onCreateContextMenu(menu, v, menuInfo)
        viewsWithShownContextMenu.add(v)
    }
}
